File: frontend/src/services/notifications.ts
import { apiGet } from "./api";

export type NotificationItem = {
  notification_id: string;
  notification_type: string;
  severity: string;
  headline: string;
  message: string;
  target_path: string;
};

export type NotificationSummaryResponse = {
  unread_count: number;
  items: NotificationItem[];
};

export function getNotifications() {
  return apiGet<NotificationSummaryResponse>("/notifications");
}

File: backend/app/schemas/notifications.py
from pydantic import BaseModel


class NotificationItem(BaseModel):
    notification_id: str
    notification_type: str
    severity: str
    headline: str
    message: str
    target_path: str
    is_read: bool


class NotificationSummaryResponse(BaseModel):
    unread_count: int
    items: list[NotificationItem]


class NotificationReadResponse(BaseModel):
    notification_id: str
    is_read: bool
    updated_utc: int